esp/bindings/http/client/httpclient.cpp

```diff
@@ -213,6 +213,13 @@ void CHttpClient::setTimeOut(unsigned int timeout)
 
 int CHttpClient::connect(StringBuffer& errmsg, bool forceNewConnection)
 {
+    if (m_socket != nullptr)
+    {
+        m_socket->close();
```

Comment:
There is similar code below when HTTPS fails, and the socket is cleaned up.  It could be put in a shared method.